src/controllers/users.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const userService = require('../services/users');
+
+const getAll = (req, res) => {
+  const users = userService.getAll();
+
+  res.send(users);
+};
+
+const getSingle = (req, res) => {
+  const { userId, spentFor, title, amount, category } = req.params;
+  const foundedUser = userService.getById(userId);
+
+  const isValidData = foundedUser
+    && typeof spentFor === 'string'
+    && typeof title === 'string'
+    && typeof amount === 'number'
+    && typeof category === 'string';
+
+  if (!isValidData) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  res.status(200);
+  res.send(foundedUser);
+};
+
+const add = (req, res) => {
+  const { name } = req.body;
+
+  if (!name) {
+    res.sendStatus(400);
```

Comment:
Replace 'res.sendStatus(400)' with 'res.status(400).json({ error: 'A detailed error message' })' to provide a more informative error response.